transport/http/client.go

```diff
@@ -85,6 +101,10 @@ func (c Client) Endpoint() endpoint.Endpoint {
 		}
 
 		resp, err := ctxhttp.Do(ctx, c.client, req)
+		for _, f := range c.after {
+			ctx = f(ctx, resp)
```

Comment:
After looking at this and comparing to Server.ServeHTTP, I'd like to see more symmetry — propose that ResponseFunc shall also return a context. Would you be willing to make that change in this PR too?